src/sage/graphs/generic_graph.py

```diff
@@ -938,6 +939,143 @@ def _latex_(self):
+
+        INPUT:
+
+        - ``format`` -- string (default: ``None``), ``'dot2tex'`` or
+          ``'tkz_graph'``. If ``None``, it is set to ``'dot2tex'`` if
+          dot2tex is present, otherwise it is set to ``'tkz_graph'``.
+        - ``edge_labels`` -- bool (default: ``None``), if ``None``
+          it is set to ``True`` if and only if format is ``'dot2tex'``
+        - ``color_by_label`` -- bool (default: ``False``)
+        - ``use_sage_preamble`` -- bool (default: ``None``), if ``None``
+          it is set to ``True`` if and only if format is ``'tkz_graph'``
+
+        When using format ``'dot2tex'``, the following inputs are considered:
+
+        - ``prog`` -- string (default: ``'dot'``) the program used for the
+          layout corresponding to one of the software of the graphviz
+          suite: 'dot', 'neato', 'twopi', 'circo' or 'fdp'.
+        - ``rankdir`` -- string (default: ``'down'``)
+        - ``subgraph_clusters`` -- (default: ``[]``) a list of lists of
+          vertices, if supported by the layout engine, nodes belonging to
+          the same cluster subgraph are drawn together, with the entire
+          drawing of the cluster contained within a bounding rectangle.
+
+        OUTPUT:
+
+        An instance of :mod:`sage.misc.latex_standalone.TikzPicture`.
+
+        .. NOTE::
+
+            Prerequisite: dot2tex optional Sage package and graphviz must be
+            installed when using format ``'dot2tex'``.
+
+        EXAMPLES::
+
+            sage: g = graphs.PetersenGraph()
+            sage: tikz = g.tikz()                              # optional - dot2tex graphviz
+            sage: _ = tikz.pdf()      # not tested
+
+        ::
+
+            sage: tikz = g.tikz(format='tkz_graph')
+            sage: _ = tikz.pdf()      # not tested
+
+        Using another value for ``prog``::
+
+            sage: tikz = g.tikz(prog='neato',        # long time (3s), optional - dot2tex graphviz
+            ....:               color_by_label=True)
+            sage: _ = tikz.pdf()      # not tested
+
+        Using another value for ``rankdir``::
+
+            sage: tikz = g.tikz(rankdir='right')     # long time (3s), optional - dot2tex graphviz
+            sage: _ = tikz.pdf()      # not tested
+
+        Using subgraphs clusters (broken when using labels, see
+        :issue:`22070`)::
+
+            sage: S = FiniteSetMaps(5)
+            sage: I = S((0,1,2,3,4))
+            sage: a = S((0,1,3,0,0))
+            sage: b = S((0,2,4,1,0))
+            sage: roots = [I]
+            sage: succ = lambda v: [v*a,v*b,a*v,b*v]
+            sage: R = RecursivelyEnumeratedSet(roots, succ)
+            sage: G = R.to_digraph()
+            sage: G
+            Looped multi-digraph on 27 vertices
+            sage: C = G.strongly_connected_components()
+            sage: tikz = G.tikz(subgraph_clusters=C)    # optional - dot2tex graphviz
+            sage: _ = tikz.pdf()      # not tested
```

Comment:
when trying this test, I get many latex compilation errors like
```
! Undefined control sequence.
<recently read> \text 
      
l.63 ... (758.0bp,23.472bp) [draw,draw=none] {$\text
                                                  {\texttt{[0,{ }0,{ }0,{ }0...

! Undefined control sequence.
<recently read> \text 
```
and it ends with
```
CalledProcessError: Command '['lualatex', '-interaction=nonstopmode', 'tikz_fxvpbw29.tex']' returned non-zero exit status 1.
```
I don't know where this error comes from (my system ? some missing instruction ?)

Reply:
There was a "\usepackage{amstext}" line missing. I think it was there by default at some point.
This is now fixed. I also made the last line tested to avoid this doctest broken again in the future.